tests/resources/serializers/test_csl_serializer.py

```diff
@@ -39,3 +59,100 @@ def test_csl_serializer(
+        es_clear, minimal_record):
+
+    service = current_rdm_records.records_service
+    superuser_identity = running_app.superuser_identity
+    # create the draft
+    draft = service.create(superuser_identity, minimal_record)
+    # publish the record
+    record = service.publish(draft.id, superuser_identity)
+    doi = record.to_dict()["pids"]["doi"]["identifier"]
+
+    # test resolution
+    resolved_record = service.resolve_pid(
+        id_=doi,
+        identity=superuser_identity,
+        pid_type="doi"
+    )
+
+    id = record.id
+    url = f"/api/records/{id}?locale=en-US&style=3d-printing-in-medicine"
+    headers = {"Accept": "text/x-bibliography"}
+    response = client.get(url, headers=headers)
+    breakpoint()
+
+    expected_data = {'abstract': 'Test', 'language': 'dan',
+                     'issued': {'date-parts': [['2018'], ['2020', '09']]},
+                     'type': 'graphic',
+                     'note': 'Funding by European Commission ROR 1234.',
+                     'version': 'v1.0', 'title': 'InvenioRDM',
+                     'author': [{'family': 'Nielsen, Lars Holm'}],
+                     'id': '12345-abcde', 'publisher': 'InvenioRDM',
+                     'DOI': '10.5281/inveniordm.1234'}
+
+    assert response == expected_data
+
+
+def test_citation_string_serializer_request_no_style(
+        running_app, full_record, vocabulary_clear, client):
+
+    id = "12345-abcde"
+    url = f"/api/records/{id}?locale=en-US&"
+    headers = "text/x-bibliography"
+    response = client.get(url, headers=headers)
+
+    expected_data = {'abstract': 'Test', 'language': 'dan',
+                     'issued': {'date-parts': [['2018'], ['2020', '09']]},
+                     'type': 'graphic',
+                     'note': 'Funding by European Commission ROR 1234.',
+                     'version': 'v1.0', 'title': 'InvenioRDM',
+                     'author': [{'family': 'Nielsen, Lars Holm'}],
+                     'id': '12345-abcde', 'publisher': 'InvenioRDM',
+                     'DOI': '10.5281/inveniordm.1234'}
+
+    assert response == expected_data
+
+
+def test_citation_string_serializer_request_wrong_style(
+        running_app, full_record, vocabulary_clear, client):
+
+    id = "12345-abcde"
+    url = f"/api/records/{id}?locale=en-US&style=3d-printing-in-medici"
+    headers = "text/x-bibliography"
+    response = client.get(url, headers=headers)
+
+    expected_data = {'abstract': 'Test', 'language': 'dan',
+                     'issued': {'date-parts': [['2018'], ['2020', '09']]},
+                     'type': 'graphic',
+                     'note': 'Funding by European Commission ROR 1234.',
+                     'version': 'v1.0', 'title': 'InvenioRDM',
+                     'author': [{'family': 'Nielsen, Lars Holm'}],
+                     'id': '12345-abcde', 'publisher': 'InvenioRDM',
+                     'DOI': '10.5281/inveniordm.1234'}
+
+    assert response == expected_data
```

Comment:
I would test also from no locale. Maybe you can put all these similar tests into one function:
```
def test_..._params(...):

   - test default params
   - test different locale param
   - test different style param
   - test wrong local param
   - test wrong style param
```